api/src/main/java/io/grpc/CallOptions.java

```diff
@@ -203,6 +207,46 @@ public CallOptions withoutWaitForReady() {
     return builder.build();
   }
 
+  /**
+   * Specifies how many bytes must be queued before the call will leave the
+   * <a href="https://github.com/grpc/grpc/blob/master/doc/wait-for-ready.md">
+   * 'wait for ready'</a> state.
+   *
+   * @param numBytes The number of bytes that must be queued. Must be a
+   *                 positive integer.
+   */
+  @ExperimentalApi("https://github.com/grpc/grpc-java/issues/11021")
+  public CallOptions withOnReadyThreshold(int numBytes) {
+    Builder builder = toBuilder(this);
```

Comment:
Add verification. `checkArgument(numBytes > 0, "numBytes must be positive: %s", numBytes)`